src/wristpy/core/cli.py

```diff
@@ -0,0 +1,100 @@
+"""CLI for wristpy."""
+
+import argparse
+import pathlib
+from typing import List, Optional
+
+from wristpy.core import config, orchestrator
+
+logger = config.get_logger()
+
+
+def parse_arguments(args: Optional[List[str]] = None) -> argparse.Namespace:
+    """Argument parser for Wristpy's cli.
+
+    Args:
+        args: A list of command line arguments given as strings. If None, the parser
+            will take the args from `sys.argv`.
+
+    Returns:
+        Namespace object with all of the input arguments and default values.
+    """
+    default_settings = config.Settings()
+    parser = argparse.ArgumentParser(
+        description="Run the main wristpy pipeline",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        epilog="Please report issues at https://github.com/childmindresearch/wristpy",
+    )
+
+    parser.add_argument("input", type=pathlib.Path, help="Path to the input data.")
+
+    parser.add_argument(
+        "-o",
+        "--output",
+        type=pathlib.Path,
+        help="Path where data will be saved. Supports .csv and .parquet formats.",
+    )
+
+    parser.add_argument(
+        "-c",
+        "--calibrator",
+        type=str,
+        choices=["ggir", "gradient", "none"],
+        default="none",
+        help="Pick which calibrator to use. Can be 'ggir' or 'gradient'.",
+    )
+    parser.add_argument(
+        "-t",
+        "--thresholds",
+        type=float,
+        nargs=3,
+        default=[
+            default_settings.LIGHT_THRESHOLD,
+            default_settings.MODERATE_THRESHOLD,
+            default_settings.VIGOROUS_THRESHOLD,
+        ],
+        help="Provide three thresholds for light, moderate, and vigorous activity."
+        "Values must be given in ascending order.",
+    )
+
+    parser.add_argument(
+        "-e",
+        "--epoch_length",
+        default=5,
+        type=int,
```

Comment:
This is correct, though @Asanto32 @frey-perez do we want to make an issue out of allowing floats? This would require updates elsewhere so beyond the scope of this PR either way.